pkg/cluster/config/validate.go

```diff
@@ -27,19 +27,23 @@ import (
 func (c *Config) Validate() error {
 	errs := []error{}
 
-	numByRole := make(map[NodeRole]int)
+	numByRole := make(map[NodeRole]int32)
 	// All nodes in the config should be valid
 	for i, n := range c.Nodes {
-		// update role count
-		if num, ok := numByRole[n.Role]; ok {
-			numByRole[n.Role] = 1 + num
-		} else {
-			numByRole[n.Role] = 1
-		}
 		// validate the node
 		if err := n.Validate(); err != nil {
 			errs = append(errs, errors.Errorf("invalid configuration for node %d: %v", i, err))
 		}
+		// update role count
+		replicas := int32(1)
+		if n.Replicas != nil {
+			replicas = *n.Replicas
```

Comment:
observation: without this patch we seem to be handling `replicas: 0` as "no nodes" of this role.
but with the planned removal of replicas this should not be a problem.

Reply:
this should be the same now? if n.Replicas is set we will use the value?

Reply:
i was just mentioning that we might want to throw errors for replicas < 1, but this is not important given the planned removal of replicas.

Reply:
ah I see, we do validate < 0, zero seems like a quasi valid value. when translating to a version without replicas we could drop this node type.